### PR TITLE
Remove internalclientset usage from oc helpers

### PR DESCRIPTION
**Pull request by juanvallejo**
Further snips dependencies from `oc` on the core repo.
See: https://docs.google.com/spreadsheets/d/19beyAMX3Jp-rlh9i68DgXVSoE9OF3C9EiLs2rPJvpic/edit#gid=0

cc @soltysh @deads2k

### CODE REVIEW

File: pkg/oc/cli/startbuild/startbuild_test.go [hunk 1]
```diff
@@ -21,13 +21,13 @@ import (
...
 	clientcmdapi "k8s.io/client-go/tools/clientcmd/api"
-	"k8s.io/kubernetes/pkg/api/legacyscheme"
 	"k8s.io/kubernetes/pkg/kubectl/genericclioptions"
+	"k8s.io/kubernetes/pkg/kubectl/scheme"
 
 	buildv1 "github.com/openshift/api/build/v1"
 	buildapi "github.com/openshift/origin/pkg/build/apis/build"
 	buildclientmanual "github.com/openshift/origin/pkg/build/client/v1"
-	"github.com/openshift/origin/pkg/oauth/generated/internalclientset/scheme"
+	"k8s.io/kubernetes/pkg/api/legacyscheme"
```
Comment (soltysh) [inline pkg/oc/cli/startbuild/startbuild_test.go]: Move up to k8s imports.

File: pkg/oc/cli/logs/logs.go [hunk 2]
```diff
@@ -20,7 +20,7 @@ import (
 	buildv1 "github.com/openshift/api/build/v1"
 	buildv1client "github.com/openshift/client-go/build/clientset/versioned/typed/build/v1"
 	buildapi "github.com/openshift/origin/pkg/build/apis/build"
-	buildutil "github.com/openshift/origin/pkg/build/util"
+	ocbuildapihelpers "github.com/openshift/origin/pkg/oc/lib/buildapihelpers"
```
Comment (soltysh) [inline pkg/oc/cli/logs/logs.go]: Honestly, I'm not a big fan, but ok.

File: pkg/oc/originpolymorphichelpers/logsforobject.go [hunk 3]
```diff
@@ -34,17 +29,6 @@ func NewLogsForObjectFn(delegate polymorphichelpers.LogsForObjectFunc) polymorph
 		}
 
 		switch t := object.(type) {
-		case *appsapi.DeploymentConfig:
```
Comment (soltysh) [inline pkg/oc/originpolymorphichelpers/logsforobject.go]: :+1:

File: pkg/oc/util/clientcmd/resolve.go [hunk 4]
```diff
@@ -53,6 +53,8 @@ func resolveImagePullSpec(imageClient imageclient.ImageInterface, source, name,
 		return "", fmt.Errorf("unable to resolve %s %q", source, name)
 	}
 
+	// TODO(juanvallejo): This helper might need to be copied into the `oc` tree in order
+	// to snip this dependency on the internal image api tree
```
Comment (soltysh) [inline pkg/oc/util/clientcmd/resolve.go]: This should be replaced with https://github.com/openshift/library-go/blob/a7d809ee41d6ca6720192824ee54ed8f19c3cfac/pkg/image/reference/reference.go#L32, if you look under the covers of `imageapi.ParseDockerImageReference` it call just that function.

File: pkg/oc/cli/logs/logs.go [hunk 5]
```diff
@@ -261,13 +261,14 @@ func (o *LogsOptions) runLogPipeline() error {
...
 	case isBC:
-		buildName := buildutil.BuildNameForConfigVersion(bc.ObjectMeta.Name, int(bc.Status.LastVersion))
+		buildName := ocbuildapihelpers.BuildNameForConfigVersion(bc.ObjectMeta.Name, int(bc.Status.LastVersion))
 		build, _ = o.Client.Builds(o.Namespace).Get(buildName, metav1.GetOptions{})
 		if build == nil {
 			return fmt.Errorf("the build %s for build config %s was not found", buildName, bc.Name)
 		}
 		fallthrough
 	case isBld:
+		// TODO(juanvallejo): what to do about this? Const not available in external api, but we need to get rid of dependency on internal api
```
Comment (soltysh) [inline pkg/oc/cli/logs/logs.go]: Some of these will have to be sniped out to API. We need to check if they are worthy, though.

File: pkg/oc/util/clientcmd/resolve.go (hunk 4 above)
Comment (juanvallejo) [reply pkg/oc/util/clientcmd/resolve.go]: That's great, didn't know about this helper. Will sweep remaining uses of the imageutil helper as well

**Pull request merged**